cmd/up_test.go

```diff
@@ -61,3 +61,27 @@ func TestIsGKE(t *testing.T) {
 		t.Errorf("expect non-GKE but got GKE")
 	}
 }
+
```

Comment:
the function `SecretExists` has a lot of if-else's, you should ensure what is the behavior you expect with unit tests

Reply:
Correct. The reason I haven't added test for this because I don't know how to create a fake client with clientPool and disco yet. I suggest to unblock this for now by just merging it. I will work later on that unit test.

Reply:
Right, unfortunately you don't get a mock-able dynamic client until client-go v5.0.  I chose to exercise most of these sorts of code paths via integration tests in the kubecfg codebase fwiw, for this reason.